Orange/widgets/evaluate/owtestlearners.py

```diff
@@ -258,28 +279,28 @@ def set_train_data(self, data):
         """
         Set the input training dataset.
         """
-        self.error(0)
-        self.information(0)
+        self.clear_messages(self.ERR_CLASS_REQUIRED, self.INFO_DATA_SAMPLED)
         if data and not data.domain.class_var:
-            self.error(0, "Train data input requires a class variable")
+            self.message(self.ERR_CLASS_REQUIRED)
             data = None
 
         if isinstance(data, SqlTable):
             if data.approx_len() < AUTO_DL_LIMIT:
                 data = Table(data)
             else:
-                self.information(0, "Train data has been sampled")
+                self.message(self.INFO_DATA_SAMPLED)
```

Comment:
Why then not just `self.INFO_DATA_SAMPLED.show()` and `self.WARN_MISSING_DATA.hide()` ?